raft/progress.go

```diff
@@ -203,6 +202,17 @@ func (in *inflights) add(inflight uint64) {
 	if next >= in.size {
 		next -= in.size
 	}
+	if next >= len(in.buffer) {
```

Comment:
can we put this into a separate func? growBuf or something. also can we doc why do we want to do this? (there are use cases that one logic node can have 1,000s of raft...)